src/main/java/jewellery/inventory/service/OrganizationService.java

```diff
@@ -67,6 +80,37 @@ private List<Organization> getAll() {
     return organizationRepository.findAll();
   }
 
+  private void validateOrganizationProductsAndResources(Organization organization) {
+    if (!organization.getProductsOwned().isEmpty()) {
+      throw new OrganizationProductsException(organization.getId());
+    }
+    if (!organization.getResourceInOrganization().isEmpty()) {
+      throw new OrganizationResourcesException(organization.getId());
+    }
+  }
+
+  public void validateCurrentUserPermission(
+      Organization organization, OrganizationPermission permission) {
+    User currentUser = userService.getUser(authService.getCurrentUser().getId());
+    if (!hasPermission(currentUser, organization, permission)) {
+      throw new MissingOrganizationPermissionException(
+          currentUser.getId(), organization.getId(), permission);
+    }
+    logger.debug(
+        "User permission validation successful. User ID: {}, Organization ID: {}",
+        currentUser.getId(),
+        organization.getId());
+  }
+
+  private boolean hasPermission(
+      User user, Organization organization, OrganizationPermission permission) {
+    return organization.getUsersInOrganization().stream()
+        .anyMatch(
+            userInOrganization ->
+                userInOrganization.getUser().equals(user)
```

Comment:
Here we can have NPE. Better to write it as:
user.equals(userInOrganization.getUser())